### PR TITLE
fix(timepicker): handle midnight properly

### PR DESCRIPTION
**Pull request by gergelyke**
Closes https://github.com/uber/baseweb/issues/2870

### CODE REVIEW

File: src/timepicker/timepicker.js [hunk 1]
```diff
@@ -160,7 +160,9 @@ class TimePicker extends React.Component<TimePickerPropsT, TimePickerStateT> {
 
   handleChange = (seconds: number) => {
     const date =
-      this.props.value && seconds ? new Date(this.props.value) : new Date();
+      this.props.value && typeof seconds === 'number'
```
Comment (chasestarr) [inline src/timepicker/timepicker.js]: Based on the type definition, it needs to be a number or flow will error. In what situation will seconds not be defined?

Comment (gergelyke) [reply src/timepicker/timepicker.js]: seconds can be zero, making it a falsy value - this change tries to cover that case

Comment (chasestarr) [reply src/timepicker/timepicker.js]: can `&& seconds` be removed?

Comment (gergelyke) [reply src/timepicker/timepicker.js]: that's a very good point - yes

Comment (gergelyke) [reply src/timepicker/timepicker.js]: @chasestarr fixed - @nadiia does this make sense to you too?

**Comment by gergelyke** [comment]
> a test would better ensure the reported bug is fixed, but logically this should work

Would you write a unit test for this?

**Comment by chasestarr** [comment]
I'd probably write an e2e test

**Comment by gergelyke** [comment]
will follow up with a test pr later this week

**Pull request merged**